flux2.tf

```diff
@@ -113,59 +127,66 @@ resource "kubernetes_secret" "main" {
   data = {
     "identity.pub" = tls_private_key.identity[0].public_key_pem
     identity       = tls_private_key.identity[0].private_key_pem
-    username       = "git"
-    password       = local.flux2["github_token"]
+    known_hosts    = join("\n", local.flux2["known_hosts"])
   }
 }
 
 # GitHub
 resource "github_repository" "main" {
-  count      = local.flux2["enabled"] && local.flux2["create_github_repository"] ? 1 : 0
+  count      = local.flux2["enabled"] && local.flux2["create_github_repository"] && (local.flux2["provider"] == "github") ? 1 : 0
   name       = local.flux2["repository"]
   visibility = local.flux2["repository_visibility"]
   auto_init  = true
 }
 
+data "github_repository" "main" {
+  count = local.flux2["enabled"] && local.flux2["create_github_repository"] && (local.flux2["provider"] == "github") ? 0 : 1
+  name  = local.flux2["repository"]
+}
+
 resource "github_branch_default" "main" {
-  count      = local.flux2["enabled"] && local.flux2["create_github_repository"] ? 1 : 0
-  repository = local.flux2["create_github_repository"] ? github_repository.main[0].name : local.flux2["repository"]
+  count      = local.flux2["enabled"] && local.flux2["create_github_repository"] && (local.flux2["provider"] == "github") ? 1 : 0
+  repository = local.flux2["create_github_repository"] ? github_repository.main[0].name : data.github_repository.main[0].name
   branch     = local.flux2["branch"]
 }
 
 resource "github_repository_deploy_key" "main" {
-  count      = local.flux2["enabled"] ? 1 : 0
+  count      = local.flux2["enabled"] && (local.flux2["provider"] == "github") ? 1 : 0
   title      = "flux-${local.flux2["create_github_repository"] ? github_repository.main[0].name : local.flux2["repository"]}-${local.flux2["branch"]}"
-  repository = local.flux2["create_github_repository"] ? github_repository.main[0].name : local.flux2["repository"]
+  repository = local.flux2["create_github_repository"] ? github_repository.main[0].name : data.github_repository.main[0].name
   key        = tls_private_key.identity[0].public_key_openssh
   read_only  = true
```

Comment:
Should we add an `auto_image_update` variable and set this to `false` if `auto_image_update` is `true` ?